spec/models/manageiq/providers/google/network_manager/refresh_parser_spec.rb

```diff
@@ -0,0 +1,186 @@
+        :ems_ref                      => "some-id",
+        :load_balancer_protocol       => "TCP",
+        :instance_protocol            => "TCP",
+        :load_balancer_port_range     => (8080..8090),
+        :instance_port_range          => (8080..8090),
+        :load_balancer                => hashes[:load_balancers][0],
+        :load_balancer_listener_pools => [
+          :load_balancer_pool         => hashes[:load_balancer_pools][0]
+        ]
+      ]
+    )
+    expect(hashes[:load_balancer_pools]).to eql(
+      [
+        :type                            => "ManageIQ::Providers::Google::NetworkManager::LoadBalancerPool",
+        :ems_ref                         => "some-target-pool-id",
+        :name                            => "my-tp",
+        :load_balancer_pool_member_pools => [
+          :load_balancer_pool_member => {
+            :type    => "ManageIQ::Providers::Google::NetworkManager::LoadBalancerPoolMember",
+            :ems_ref => Digest::MD5.base64digest(
+              "https://www.googleapis.com/compute/v1/projects/#{ems.project}/zones/#{az.name}/instances/#{vm.name}"),
+            :vm      => vm
+          }
+        ]
+      ]
+    )
+  end
+
+  describe '#parse_port_range' do
+    it 'treats a single port as a single port range' do
+      expect(described_class.parse_port_range("80")).to eql(80..80)
+    end
+
+    it 'treats port range as a port range' do
+      expect(described_class.parse_port_range("100-123")).to eql(100..123)
+    end
+
+    it 'treats an empty string as the entire tcp/udp port range' do
+      expect(described_class.parse_port_range("")).to eql(0..65_535)
+    end
+  end
+
+  describe 'described_class#parse_vm_link' do
+    it 'does not match invalid link urls' do
+      expect(described_class.parse_vm_link("")).to eql(nil)
+      expect(
+        described_class.parse_vm_link(
+          "https://example.com/compute/v1/projects/foo/zones/bar/instances/baz")).to eql(nil)
+      expect(
+        described_class.parse_vm_link(
+          "https://www.googleapis.com/compute/v1/projects/foo/zones/bar/instances/baz/bam/boom")).to eql(nil)
+    end
+
+    it 'does match valid link url' do
+      vm_link = "https://www.googleapis.com/compute/v1/projects/foo/zones/bar/instances/baz"
+      expect(described_class.parse_vm_link(vm_link)).to eql(
+        :project  => "foo",
+        :zone     => "bar",
+        :instance => "baz"
+      )
+    end
+  end
+
+  private
+
+  def set_forwarding_rules(connection, items)
+    allow(connection).to receive(:forwarding_rules) { forwarding_rules_collection(items) }
+  end
+
+  def set_target_pools(connection, items)
+    allow(connection).to receive(:target_pools) { target_pools_collection(items) }
+  end
+
+  def forwarding_rules_collection(items)
+    fog_collection(items, :name)
+  end
+
+  def target_pools_collection(items)
+    fog_collection(items, :name)
+  end
```

Comment:
Yeah, I think you're right; initially I had about 5-6 different collections which had different semantics for `get`, but then I spent time cleaning it up so there's only those two. In the interest of keeping it readable and simple, I'll nuke these helper methods.